model/build/task_cache.go

```diff
@@ -71,11 +72,11 @@ func SetCachedTaskStarted(buildId, taskId string, startTime time.Time) error {
 
 // SetCachedTaskFinished sets the given task to "finished"
 // along with a time taken in the cache of the given build.
-func SetCachedTaskFinished(buildId, taskId string, detail *apimodels.TaskEndDetail, timeTaken time.Duration) error {
+func SetCachedTaskFinished(buildId, taskId string, detail *apimodels.TaskEndDetail, status string, timeTaken time.Duration) error {
```

Comment:
does it make sense to reorder the args to put status before the task end detail.

Reply:
switched the order